geoapps_utils/driver/data.py

```diff
@@ -0,0 +1,50 @@
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps-utils package.
+#
+#  All rights reserved.
+#
+
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps-utils.
+#
+#  geoapps-utils is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from pathlib import Path
+from typing import Union
+
+from geoh5py.workspace import Workspace
+from pydantic import BaseModel
+
+
+class BaseData(
+    BaseModel,
+    arbitrary_types_allowed=True,  # type: ignore
+    extra="ignore",  # type: ignore
```

Comment:
**question:** why ignoring type checks here?
```suggestion
    arbitrary_types_allowed: bool=True,
    extra:str ="ignore",
```

Reply:
These were ignoring an unexpected keyword error I was getting. I've removed these now and added a pydantic plugin to the pyproject.toml to fix the error.